core/build.gradle

```diff
@@ -24,8 +24,15 @@ import org.elasticsearch.gradle.test.RestSpecHack
 apply plugin: 'elasticsearch.build'
 apply plugin: 'com.bmuschko.nexus'
 apply plugin: 'nebula.optional-base'
+apply plugin: "application"
 
 archivesBaseName = 'elasticsearch'
+def runEsHome="${project.buildDir}/run-elastic"
+mainClassName = "org.elasticsearch.bootstrap.Elasticsearch"
+applicationDefaultJvmArgs = ["-Xms256m", "-Xmx1g", "-Djava.awt.headless=true", "-Delasticsearch", "-Des.foreground=yes",
+                             "-ea", "-Des.path.home=" + runEsHome, "-Des.security.manager.enabled=false",
```

Comment:
`run.sh` was good because it was "real". It run with the security manager and with `bin/elasticsearch`. So much real. Sadly this sacrifices a lot.